xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -320,13 +320,6 @@ EndpointUpdate build() {
   void watchConfigData(String hostName, int port, ConfigWatcher watcher) {
```

Comment:
May be this was discussed before, but why not have the caller just pass the hostPort string (authority) so the caller doesn't need to parse into host+port and then pass -1 for unspecified etc. Seems much cleaner?

Reply:
You would need two strings: "hostName" and "hostName:port". So either way, if you pass in "hostName:port" you need to break it up and if you pass in hostName and port, you need to concatenate. The later is easier.

Reply:
No, you just require `hostName:port` with port being optional. So you don't expect the caller to pass -1 for not specified etc. Just looks cleaner but I have no strong opinion.

Reply:
> you just require `hostName:port` with port being optional

It also requires `hostname`, see the line that propagate the field
```
this.hostName = checkNotNull(hostName, "hostName");
```
which looks a little awkward.

Reply:
Okay but you can parse hostName out of "hostName:port" : that was my point